packages/flutter_test/lib/src/binding.dart

```diff
@@ -1044,14 +1044,29 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
   @override
   void _verifyInvariants() {
     super._verifyInvariants();
-    assert(
-      _currentFakeAsync.periodicTimerCount == 0,
-      'A periodic Timer is still running even after the widget tree was disposed.'
-    );
-    assert(
-      _currentFakeAsync.nonPeriodicTimerCount == 0,
-      'A Timer is still pending even after the widget tree was disposed.'
-    );
+
+    assert(() {
+      if (   _currentFakeAsync.periodicTimerCount == 0
+          && _currentFakeAsync.nonPeriodicTimerCount == 0) {
+        return true;
+      }
+
+      debugPrint('Pending timers:');
+      for (String timerInfo in _currentFakeAsync.pendingTimersDebugInfo) {
+        final int firstLineEnd = timerInfo.indexOf('\n');
+        assert(firstLineEnd != -1);
+
+        // No need to include the newline.
+        final String firstLine = timerInfo.substring(0, firstLineEnd);
+        final String stackTrace = timerInfo.substring(firstLineEnd + 1);
+
+        debugPrint(firstLine);
+        debugPrintStack(stackTrace: StackTrace.fromString(stackTrace));
+        debugPrint('');
+      }
+      return false;
+    }(),'A Timer is still pending even after the widget tree was disposed.');
```

Comment:
trivialiest of nits: space missing after comma

Reply:
Oops.  Fixed.

> Can we add a test for this? In the past I've found this kind of code ends up being very brittle and because it only gets hit when things fail it takes a long time for anyone to notice it broke.

Okay, I'll see if I can make a relevant test.

Reply:
I remember now why I didn't add a test: I don't know how.  I want to test that a test fails, but `test`/`testWidgets` doesn't allow nesting.  I assume that my test will need to essentially run `flutter test` on a failing test and interpret the ouptut.

Do you know of examples of tests of tests that I can use for reference?